src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -2,13 +2,16 @@
 
 import static de.tum.in.www1.artemis.config.Constants.MAX_NUMBER_OF_LOCKED_SUBMISSIONS_PER_TUTOR;
 
+import org.springframework.stereotype.Service;
+
 import de.tum.in.www1.artemis.domain.Exercise;
 import de.tum.in.www1.artemis.domain.StudentParticipation;
 import de.tum.in.www1.artemis.domain.Submission;
 import de.tum.in.www1.artemis.repository.SubmissionRepository;
 import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
 
-abstract class SubmissionService {
+@Service
+public class SubmissionService {
```

Comment:
I would suggest to leave it abstract and have specific implementation for ProgrammingSubmissionService as we have already for other exercise types

Reply:
From my point of view composition is preferable to inheritance. I'm closing this for now but we might want to come back to discussing this.